controllers/repository_test.go

```diff
@@ -122,3 +123,118 @@ func TestListRepositories(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestGetImage(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockECR := mock.NewMockECRAPI(ctrl)
+	controller := NewRepositoryController(mockECR)
+
+	validateDescribeImagesInput := func(input *ecr.DescribeImagesInput) {
+		if v, want := aws.StringValue(input.RepositoryName), "user/test"; v != want {
+			t.Errorf("Name was '%v', expected '%v'", v, want)
+		}
+		if v, want := aws.StringValue(input.ImageIds[0].ImageTag), "test"; v != want {
+			t.Errorf("Tag was '%v', expected '%v'", v, want)
+		}
+	}
+
+	tag := ""
+
+	tags := []*string{}
+	tags = append(tags, &tag)
+
+	detail := &ecr.ImageDetail{}
+	detail.SetImageDigest("")
```

Comment:
Once you add the safe de-reference functions, you can get rid of this extra detail init stuff (we only care about setting the tag field)